src/main/java/org/opentripplanner/middleware/controllers/api/ApiUserController.java

```diff
@@ -84,13 +99,24 @@ private boolean userHasKey(ApiUser user, String apiKeyId) {
                 .anyMatch(apiKey -> apiKeyId.equals(apiKey.keyId));
     }
 
+    /**
+     * Authenticate user with Auth0 based on username (email) and password. If successful, return the bearer token else
+     * null.
+     */
+    private String authenticateAuth0User(Request req, Response res) throws JsonProcessingException {
+        String username = HttpUtils.getRequiredQueryParamFromRequest(req, USERNAME_PARAM, false);
+        // FIXME: Should this be encrypted?!
+        String password = HttpUtils.getRequiredQueryParamFromRequest(req, PASSWORD_PARAM, false);
+        return Auth0Users.getAuth0Token(username, password);
```

Comment:
Here we should probably return the full JSON response that `getAuthToken` gets from its HTTPS oauth call.

Reply:
Updated.